lib/puppet/ssl/state_machine.rb

```diff
@@ -260,10 +263,12 @@ def ensure_client_certificate
   private
 
   def run_machine(state, stop)
-    loop do
-      state = state.next_state
+    lock do
+      loop do
+        state = state.next_state
 
-      return state if state.is_a?(stop)
+        return state if state.is_a?(stop)
```

Comment:
I hate the nuances of `return` and blocks so much. >_>

Is this going to return from `loop`? `lock`? `run_machine`? Mostly I'm concerned about the `return` interfering with any lock cleanup code.

Reply:
Yeah... the lock code has an ensure block to cleanly unlock, but I hear what you're saying. Let me change that to a break instead.